packages/docs/versioned_docs/version-2.0.0-alpha-3/protocol/03-epoch-key.md

```diff
@@ -4,18 +4,19 @@ description: Definition of epoch key in UniRep
 
 # Epoch Key
 
-* _Epoch keys_ are temporary personas which users use them to interact with others.
-* Instead of giving attestations to an `identityCommitment`, a random-value-like `epochKey` is the **receiver** of an attestation.
-* Epoch key is computed by
+* Unique epoch keys are generated for each user in each epoch.
+* They are the users' temporary personas, allowing anonymous interaction with others.
+* Instead of giving attestations to an `identityCommitment`, an ever-changing, random-value-like `epochKey` is the **receiver** of an attestation.
+* An epoch key is computed by:
 
 ```typescript
 hash(identityNullifier, atteserId, epoch, nonce) % 2^epochTreeDepth
 ```
 
-where `nonce` and be any value between `0` and `numEpochKeyNoncePerEpoch - 1`, so that a user can have `numEpochKeyNoncePerEpoch` epoch keys per epoch.
+where `nonce` can be any value between `0` and `numEpochKeyNoncePerEpoch - 1`, so that a user can have `numEpochKeyNoncePerEpoch` epoch keys per epoch.
```

Comment:
Is numEpochKeyNoncePerEpoch defined by each attester?

Reply:
It's defined in the proving key configuration. Each attester must use the values the keys were generated with.

Reply:
ok. So the current UniRep contract is configured with 3 epoch keys, and all attesters signed up to this contract will give users 3 epoch keys per epoch?

Reply:
yes